### PR TITLE
Fixed Issues With EVO

### LINKED ISSUE #447: Update EVO to work with restructure of data objects
As the structure of our data changes, EVO will need to be updated in order to properly access the parameters needed. This will happen primarily in the evoObjects.js file. If the structure of the analysisResult data changes as well, there will be additional updating needed. This will mostly just mean changing lines like model.getIntentionByNodeID() to cell.get('intention'), etc.

### PR DESCRIPTION
**Pull request by Manarnz**
This fixes the list of the malfunctioning methods in issue #447

### CODE REVIEW

File: leaf-ui/js/object/evoObjects.js [hunk 1]
```diff
@@ -330,21 +341,19 @@ class EVO {
...
       */
-    static colorIntentionsAnalysis()
+    static colorIntentionsAnalysis(analysisResult)
     {
         var elements = graph.getElements(); 
         var actorBuffer = 0;
     
         for (var i = 0; i < elements.length; i++) { 
-            ++count;
+            //++count;
```
Comment (YesugenB) [inline leaf-ui/js/object/evoObjects.js]: Delete comment if you don't need it

File: leaf-ui/js/object/evoObjects.js [hunk 2]
```diff
@@ -330,21 +341,19 @@ class EVO {
...
             var cellView  = elements[i].findView(paper);
-            var intention = model.getIntentionByID(cellView.model.attributes.nodeID);
-    
+            var intention = elements[i].get('intention')
```
Comment (YesugenB) [inline leaf-ui/js/object/evoObjects.js]: Add `;` at the end?

File: leaf-ui/js/object/evoObjects.js [hunk 3]
```diff
@@ -377,20 +386,22 @@ class EVO {
...
-
+            intention = elements[i].get('intention');
+            initSatVal = intention.getUserEvaluationBBM(0).get('assignedEvidencePair'); 
+            if(analysisResult !== null){
+                if(!analysisResult.get('isPathSim')){   
+                    if (initSatVal === '(no value)') {
+                        curr.attr('.satvalue/text', '');
+                        curr.attr({text: {fill: 'black',stroke:'none','font-weight' : 'normal'}});
+    
+                    }else{
```
Comment (YesugenB) [inline leaf-ui/js/object/evoObjects.js]: Delete extra line above here and add a space between `else` and `}{`

Same for line 401

File: leaf-ui/js/object/evoObjects.js [hunk 4]
```diff
@@ -377,20 +386,22 @@ class EVO {
...
+                        curr = elements[i].findView(paper).model;
+                        curr.attr({text: {fill: 'white',stroke:'none'}});
+                    }
                 }else{
                     curr = elements[i].findView(paper).model;
-                    curr.attr({text: {fill: 'white',stroke:'none'}});
+                    curr.attr({text: {fill: 'white',stroke:'none'}});
```
Comment (YesugenB) [inline leaf-ui/js/object/evoObjects.js]: Delete extra space at end of line

File: leaf-ui/js/object/evoObjects.js [hunk 5]
```diff
@@ -412,10 +423,10 @@ class EVO {
     static colorIntentionsModeling(){
         var elements = graph.getElements();
         for (var i = 0; i < elements.length; i++){ 
-            var cellView = elements[i].findView(paper); 
-            var intention = model.getIntentionByID(cellView.model.attributes.nodeID); //aquires current intention
+            var cellView = elements[i].findView(paper);
+            var intention = elements[i].get('intention'); //aquires current intention
```
Comment (YesugenB) [inline leaf-ui/js/object/evoObjects.js]: Should `aquires` be `Acquires`

Also I know these comments are old but can we update them to have a space after the `//` and begin with a capital letter?

File: leaf-ui/js/onFunctions.js [hunk 6]
```diff
@@ -529,79 +460,163 @@ paper.on("link:options", function (cell) {
...
-                    selectedResult = result.name; // Record the name of result
+
+        // Individually creates each ConfigBBM and add to collection
+        for(let config of loadedConfig){
+            if (config.selected){ // If selected is true
+                selectedConfig = config.name; //Record the name of config
+            }
+            var configBBM = new ConfigBBM({name:config.name, action: config.action, conflictLevel: config.conflictLevel, numRelTime: config.numRelTime, currentState: config.currentState, userAssignmentsList : config.userAssignmentsList, previousAnalysis: config.previousAnalysis, selected: config.selected})
+            if (config.results.length !== 0){ //create results if there applicable
+                var results = configBBM.get('results') // grabs the coolection from the configbbm
```
Comment (YesugenB) [inline leaf-ui/js/onFunctions.js]: `;` at the end, `collection` is spelled wrong, and can we change `configbbm` to `configBBM`?
Also capitalize the start of the comments?

File: leaf-ui/js/onFunctions.js [hunk 7]
```diff
@@ -529,79 +460,163 @@ paper.on("link:options", function (cell) {
...
+                for (let result of config.results){
+                    if (result.selected){ // If selected is true
+                        selectedResult = result.name; // Record the name of result
+                    }
+                    var resultsBBM = new ResultBBM({name: result.name, assignedEpoch: result.assignedEpoch, timePointPath: result.timePointPath, elementList: result.elementList, allSolution: result.allSolution, isPathSim: result.isPathSim, colorVis: result.colorVis, selectedTimePoint: result.selectedTimePoint, selected: result.selected});
+                    results.add(resultsBBM)
                 }
-                var resultsBBM = new ResultBBM({name: result.name, assignedEpoch: result.assignedEpoch, timePointPath: result.timePointPath, elementList: result.elementList, allSolution: result.allSolution, isPathSim: result.isPathSim, colorVis: result.colorVis, selectedTimePoint: result.selectedTimePoint, selected: result.selected});
-                results.add(resultsBBM)
+                configCollection.add(configbbm)
```
Comment (YesugenB) [inline leaf-ui/js/onFunctions.js]: Should it be `configBBM`? Also, it looks like ConfigBBM is being added to the collection twice(lines 518 and 520). Is this right?

File: leaf-ui/js/onFunctions.js [hunk 8]
```diff
@@ -529,79 +460,163 @@ paper.on("link:options", function (cell) {
...
 
-    var currResult;
-    if (configGroup[0].get('results').length !== 0){ // Within that selected config
-        // Set selected of the selected result as true
-        currResult= configGroup[0].get('results').filter(selectedRes => selectedRes.get('name') == selectedResult)[0]
-        currResult.set('selected', true)
+        var currResult;
+        if (configGroup[0].get('results').length !== 0){ // Within that selected config
+            // Set selected of the selected result as true
+            currResult= configGroup[0].get('results').filter(selectedRes => selectedRes.get('name') == selectedResult)[0]
```
Comment (YesugenB) [inline leaf-ui/js/onFunctions.js:537]: `;`

File: leaf-ui/js/onFunctions.js [hunk 9]
```diff
@@ -529,79 +460,163 @@ paper.on("link:options", function (cell) {
...
+            currResult.set('selected', true)
```
Comment (YesugenB) [inline leaf-ui/js/onFunctions.js]: `;`

File: leaf-ui/js/onFunctions.js [hunk 10]
```diff
@@ -529,79 +460,163 @@ paper.on("link:options", function (cell) {
...
+    });
+
+    // Save the current graph to json file
+    $('#btn-save').on('click', function () {
+        var name = window.prompt("Please enter a name for your file. \nIt will be saved in your Downloads folder. \n.json will be added as the file extension.", "<file name>");
+        if (name) {
+            clearCycleHighlighting(selectResult);
+            EVO.deactivate();
+        // EVO.returnAllColors(graph.getElements(), paper);
+        // EVO.revertIntentionsText(graph.getElements(), paper);
```
Comment (YesugenB) [inline leaf-ui/js/onFunctions.js]: Can these commented out lines be removed?

File: leaf-ui/js/onFunctions.js [hunk 11]
```diff
@@ -529,79 +460,163 @@ paper.on("link:options", function (cell) {
...
+                cell.attr(".satvalue/text", "");
+                cell.attr(".funcvalue/text", "");
+
+                elementInspector.$('#init-sat-value').val('(no value)');
+                elementInspector.$('.function-type').val('(no value)');
+            }
+        }
+        IntentionColoring.refresh(selectResult);
+    });
+
```
Comment (YesugenB) [inline leaf-ui/js/onFunctions.js:582]: Delete extra line

File: leaf-ui/js/onFunctions.js [hunk 12]
```diff
@@ -529,79 +460,163 @@ paper.on("link:options", function (cell) {
...
+    /**
+     * Four option analysis mode slider
+     */
+    document.getElementById("colorResetAnalysis").oninput = function () { //changes slider mode and refreshes
+        var selectConfig;
+        //TODO: Find out why the selectResult is empty before we reassign it
+        if (configCollection.length !== 0){
+            selectConfig = configCollection.filter(Config => Config.get('selected') == true)[0];
+            if (selectConfig.get('results') !== undefined){
+                selectResult = selectConfig.get('results').filter(resultModel => resultModel.get('selected') == true)[0]
```
Comment (YesugenB) [inline leaf-ui/js/onFunctions.js]: `;`

File: leaf-ui/js/onFunctions.js [hunk 13]
```diff
@@ -529,79 +460,163 @@ paper.on("link:options", function (cell) {
...
+    // Save the current graph to json file
+    $('#btn-save').on('click', function () {
+        var name = window.prompt("Please enter a name for your file. \nIt will be saved in your Downloads folder. \n.json will be added as the file extension.", "<file name>");
+        if (name) {
+            clearCycleHighlighting(selectResult);
+            EVO.deactivate();
+        // EVO.returnAllColors(graph.getElements(), paper);
+        // EVO.revertIntentionsText(graph.getElements(), paper);    
+            var fileName = name + ".json";
+            obj = {graph: graph.toJSON()} //same structure as the other two save options
```
Comment (YesugenB) [inline leaf-ui/js/onFunctions.js]: `;` here?

**Review by YesugenB — APPROVED** [review]
Looks good! Couldn't fully test with simulate single path but from what we can tell it's working pretty smoothly

Just some small comments

**Review by i-m-foster — APPROVED** [review]

**Pull request merged**